### PR TITLE
fix(suite): make dropdown more elevated in onboarding

### PR DESCRIPTION
**Pull request by peter-sanderson**
Fixes few UI glitches in the oboarding (select backup type)

### CODE REVIEW

File: packages/suite/src/views/onboarding/steps/SelectBackupType.tsx [hunk 1]
```diff
@@ -130,13 +132,24 @@ const OptionGroupHeading = styled.div`
...
     gap: ${spacingsPx.md};
     align-items: center;
     cursor: pointer;
+
+    ${({ $hasHoverInteraction }) =>
+        $hasHoverInteraction === true
+            ? css`
+                  &:hover {
+                      background-color: ${({ theme }) => theme.backgroundSurfaceElevation2};
+                      margin: -${spacingsPx.sm};
```
Comment (peter-sanderson) [inline packages/suite/src/views/onboarding/steps/SelectBackupType.tsx]: temporary hack just to see the UI, maybe we can figure out more clear way @jvaclavik ?

File: packages/suite/src/views/onboarding/steps/SelectBackupType.tsx [hunk 2]
```diff
@@ -189,7 +202,7 @@ type OptionProps = {
 };
 
 const Option = ({ children, onSelect, isChecked, 'data-test': dataTest }: OptionProps) => (
-    <OptionStyled onClick={onSelect}>
+    <OptionStyled onClick={onSelect} $hasHoverInteraction={true}>
```
Comment (jvaclavik) [inline packages/suite/src/views/onboarding/steps/SelectBackupType.tsx:227]: Why not to have hover interaction by defaut turned on?

File: packages/suite/src/views/onboarding/steps/SelectBackupType.tsx (hunk 1 above)
Comment (jvaclavik) [reply packages/suite/src/views/onboarding/steps/SelectBackupType.tsx]: Not great, but totally understand the use case.

File: packages/suite/src/views/onboarding/steps/SelectBackupType.tsx [hunk 3]
```diff
@@ -328,14 +344,16 @@ const FloatingSelections = forwardRef<HTMLDivElement, FloatingSelectionsProps>(
                         data-test="@onboarding/select-seed-type-shamir-advance"
```
Comment (jvaclavik) [inline packages/suite/src/views/onboarding/steps/SelectBackupType.tsx:372]: What about creating this repeated structure as a separated component? 

```
<Option
                        onSelect={() => onSelect('shamir-default')}
                        isChecked={selected === 'shamir-default'}
                        data-test="@onboarding/select-seed-type-shamir-default"
                    >
                        <OptionText>
                            <Row alignItems="center">
                                <Text
                                    variant={selected === 'shamir-default' ? undefined : 'tertiary'}
                                    typographyStyle={isMobileLayout ? 'highlight' : 'titleSmall'}
                                >
                                    <Translation id={typesToLabelMap['shamir-default']} />
                                </Text>
                                {defaultType === 'shamir-default' ? (
                                    <DefaultTag />
                                ) : (
                                    <UpgradableToMultiTag />
                                )}
                            </Row>

                            <Text typographyStyle="hint">
                                <Translation id="TR_ONBOARDING_SEED_TYPE_SINGLE_SEED_DESCRIPTION" />
                            </Text>
                        </OptionText>
                    </Option>
```

File: packages/suite/src/views/onboarding/steps/SelectBackupType.tsx [hunk 4]
```diff
@@ -419,11 +441,11 @@ export const SelectBackupType = ({
         open: isOpen,
         onOpenChange: setIsOpen,
         middleware: [
-            offset(-(isMobileLayout ? SELECT_ELEMENT_HEIGHT_MOBILE : SELECT_ELEMENT_HEIGHT)),
+            offset(-(isMobileLayout ? SELECT_ELEMENT_HEIGHT_MOBILE : SELECT_ELEMENT_HEIGHT) + 1),
             size({
                 apply: ({ rects, elements, availableHeight }) => {
                     Object.assign(elements.floating.style, {
-                        width: `${rects.reference.width}px`,
+                        width: `${rects.reference.width - 2}px`,
```
Comment (jvaclavik) [inline packages/suite/src/views/onboarding/steps/SelectBackupType.tsx:478]: What is -2 (and +1 up here)?

Comment (peter-sanderson) [reply packages/suite/src/views/onboarding/steps/SelectBackupType.tsx:478]: its there for appearance of the underlying select-element



image

File: packages/suite/src/views/onboarding/steps/SelectBackupType.tsx (hunk 2 above)
Comment (peter-sanderson) [reply packages/suite/src/views/onboarding/steps/SelectBackupType.tsx:227]: This is used for "selected" element as well. We dont want it there.

**Review by jvaclavik — APPROVED** [review]

**Pull request merged**